donjo-app/models/Keuangan_grafik_manual_model.php

```diff
@@ -0,0 +1,812 @@
+		  $data['pembiayaan_keluar'][$i]['sub_pembiayaan_keluar'] = $this->get_subval_pembiayaan_keluar( $p['Akun'], $thn);
+		}
+		return $data;
+  }
+
+  private function pagu_akun($akun, $thn)
+  {
+		$this->db->select('LEFT(Kd_Rincian, 2) AS Akun, SUM(Nilai_Anggaran) AS pagu');
+		$this->db->like('Kd_Rincian', $akun, 'after');
+		$this->db->where('Tahun', $thn);
+		$this->db->group_by('Akun');
+		return $this->db->get('keuangan_manual_rinci')->result_array();
+  }
+
+  private function pagu_akun_bidang($akun, $thn)
+  {
+    $this->db->select('LEFT(Kd_Keg, 10) AS Akun, SUM(Nilai_Anggaran) AS pagu');
+		$this->db->like('LEFT(Kd_Keg, 10)', $akun, 'after');
+		$this->db->where('Tahun', $thn);
+		$this->db->group_by('Akun');
+		return $this->db->get('keuangan_manual_rinci')->result_array();
+  }
+
+  private function real_akun_pendapatan($akun, $thn=false)
+  {
+		$this->db->select('LEFT(Kd_Rincian, 2) AS Akun, SUM(Nilai_Realisasi) AS realisasi');
+		$this->db->like('Kd_Rincian', $akun, 'after');
+		$this->db->where('keuangan_manual_rinci.Tahun', $thn);
+		$this->db->group_by('Akun');
+		return $this->db->get('keuangan_manual_rinci')->result_array();
+  }
+
+  private function real_akun_pendapatan_bunga($akun, $thn=false)
+  {
+		$this->db->select('LEFT(Kd_Rincian, 2) AS Akun, SUM(Nilai_Realisasi) AS realisasi');
+		$this->db->like('Kd_Rincian', $akun, 'after');
+		$this->db->where('Tahun', $thn);
+		$this->db->group_by('Akun');
+		return $this->db->get('keuangan_manual_rinci')->result_array();
+  }
+
+  private function real_akun_belanja($akun, $thn=false)
+  {
+		$this->db->select('LEFT(Kd_Rincian, 2) AS Akun, SUM(Nilai_Realisasi) AS realisasi');
+		$this->db->like('Kd_Rincian', $akun, 'after');
+		$this->db->where('keuangan_manual_rinci.Tahun', $thn);
+		$this->db->group_by('Akun');
+		return $this->db->get('keuangan_manual_rinci')->result_array();
+  }
+
+  private function real_akun_belanja_bidang($akun, $thn=false)
+  {
+		$this->db->select('LEFT(Kd_Keg, 10) AS Akun, SUM(Nilai_Realisasi) AS realisasi');
+    $this->db->like('LEFT(Kd_Keg, 10)', $akun, 'after');
+		$this->db->where('keuangan_manual_rinci.Tahun', $thn);
+		$this->db->group_by('Akun');
+		return $this->db->get('keuangan_manual_rinci')->result_array();
+  }
+
+  private function real_akun_pembiayaan($akun, $thn=false)
+  {
+		$this->db->select('LEFT(Kd_Rincian, 2) AS Akun, SUM(Nilai_Realisasi) AS realisasi');
+		$this->db->like('Kd_Rincian', $akun, 'after');
+		$this->db->where('keuangan_manual_rinci.Tahun', $thn);
+		$this->db->group_by('Akun');
+		return $this->db->get('keuangan_manual_rinci')->result_array();
+  }
+
+  private function real_akun_pembiayaan_keluar($akun, $thn=false)
+  {
+		$this->db->select('LEFT(Kd_Rincian, 2) AS Akun, SUM(Nilai_Realisasi) AS realisasi');
+		$this->db->like('Kd_Rincian', $akun, 'after');
+		$this->db->where('keuangan_manual_rinci.Tahun', $thn);
+		$this->db->group_by('Akun');
+		return $this->db->get('keuangan_manual_rinci')->result_array();
+  }
```

Comment:
Apa beda kedua ini?

Reply:
Kedua fungsi itu berbeda karena pada fungsi akhir akan berbeda jenis kelompok yang dipakai untuk nila Realisasi nya

Pembiayaan Masuk
real_pembiayaan2 -> sub_pembiayaan2 : $this->db->where('Kelompok', '6.1.');

Pembiayaan Keluar
real_pembiayaan_keluar2 -> sub_pembiayaan_keluar2 : $this->db->where('Kelompok', '6.2.');

Reply:
Tapi functionnya persis sama -- saya lihat isi functionnya tidak ada perbedaan sama sekali? Kenapa tidak gunakan function yg sama saja -- kalau hanya paramternya yg beda -- atau hanya cara pemanggilannya saja yg beda?

Reply:
Atau bahkan sepertinya hanya ada tambahan perbedaan kelompok SESUDAH functionnya dipanggil. Kalau memang begitu, tidak perlu ada dua function yg hanya berbeda nama saja.

Reply:
Kalau misalnya script diubah menjadi sbb (yaitu sama2 menggunakan `real_akun_pembiayaan()`, apakah akan salah?
```
		$this->db->select('Akun, Nama_Akun');
		$this->db->where("Akun = '6.'");

		$data['pembiayaan'] = $this->db->get('keuangan_manual_ref_rek1')->result_array();
		foreach ($data['pembiayaan'] as $i => $p)
		{
		  $data['pembiayaan'][$i]['anggaran'] = $this->pagu_akun($p['Akun'], $thn);
		  $data['pembiayaan'][$i]['realisasi'] = $this->real_akun_pembiayaan($p['Akun'], $thn);
		  $data['pembiayaan'][$i]['sub_pembiayaan'] = $this->get_subval_pembiayaan( $p['Akun'], $thn);
		}

		$this->db->select('Akun, Nama_Akun');
		$this->db->where("Akun = '6.'");

		$data['pembiayaan_keluar'] = $this->db->get('keuangan_manual_ref_rek1')->result_array();
		foreach ($data['pembiayaan_keluar'] as $i => $p)
		{
		  $data['pembiayaan_keluar'][$i]['anggaran'] = $this->pagu_akun($p['Akun'], $thn);
		  $data['pembiayaan_keluar'][$i]['realisasi'] = $this->real_akun_pembiayaan($p['Akun'], $thn);
		  $data['pembiayaan_keluar'][$i]['sub_pembiayaan_keluar'] = $this->get_subval_pembiayaan_keluar( $p['Akun'], $thn);
```		}

Reply:
Jadi seperti function `pagu_akun()` -- yg sama untuk untuk pembiayaan maupun pembiayaan_keluar.....

Reply:
Gabung fungsi real_akun_pembiayaan_keluar dengan real_akun_pembiayaan https://github.com/OpenSID/OpenSID/pull/3119/commits/b24ab94b945812313ca34cf1495a560e5e0f5a94